GreedyParameterPointSelector.h

```diff
@@ -0,0 +1,286 @@
+       *
+       * @param[in]
+       *
+       * @return The index of the point in the list of parameters, or -1 if a
+       *         residual is not currently required.
+       */
+      int
+      getNextPointRequiringResidual();
+
+      /**
+       * @brief Set the residual error of the specified parameter point.
+       *
+       * @return The index of the point in the list of parameters.
+       */
+      void
+      setPointResidual(double error, int rank, int num_procs);
+
+      /**
+       * @brief Returns the index to the nearest local ROM to the specified parameter point.
+       *
+       * @return The index of the point in the list of parameters.
+       */
+      int
+      getNearestROM(int index);
+
+      /**
+       * @brief Get the domain of the parameter points.
+       */
+      std::vector<Vector>
+      getParameterPointDomain();
+
+      /**
+       * @brief Print the sampled parameter points to a file.
+       */
+      void
+      writeSampledPoints(std::string const& path);
+
+  private:
+
+      void constructObject(
+          std::vector<Vector> parameter_points,
+          double tolerance,
+          double saturation,
+          int subset_size,
+          int convergence_subset_size,
+          bool use_centroid,
+          int random_seed,
+          bool debug_algorithm);
+
+      /**
+       * @brief The parameter points to explore.
+       */
+      std::vector<Vector> d_parameter_points;
+
+      /**
+       * @brief The parameter points that were already sampled.
+       */
+      std::set<int> d_parameter_sampled_indices;
+
+      /**
+       * @brief The parameter point indices (used to generate the random subsets)
+       */
+      std::vector<int> d_parameter_point_random_indices;
+
+      /**
+       * @brief The current errors of the parameter points
+       */
+      std::vector<double> d_parameter_point_errors;
+
+      /**
+       * @brief The current max error of the parameter points of the current iteration
+       */
+      double d_max_error;
+
+     /**
+      * @brief The convergence tolerance used to terminate the greedy procedure.
+      */
+      double d_tol;
+
+     /**
+      * @brief The saturation constant
+      */
+      double d_sat;
+
+     /**
+      * @brief The size of the subset of parameter points used per iteration.
+      */
+      int d_subset_size;
+
+     /**
+      * @brief The size of the subset of parameter points used to check convergence.
+      */
+      int d_convergence_subset_size;
+
+
+     /**
+      * @brief The next parameter point to sample.
+      */
+      int d_next_point_to_sample;
+
+     /**
+      * @brief The next parameter point requiring a residual.
+      */
+      int d_next_point_requiring_residual;
+
+     /**
+      * @brief Whether the use the centroid heuristic for obtaining the first
+      *        parameter point.
+      */
+      bool d_use_centroid;
+
+      /**
```

Comment:
Indenting increases here. It doesn't need to be fixed. I just want to point out that having `make style` in libROM would be nice.

Reply:
I do plan to add a 'make style' using astyle, but I think it would be better when there are no open PR's as I tried it once before and it ended up with a bunch of complicated merge conflicts, so I reverted.